stac_fastapi/pgstac/app.py

```diff
@@ -64,17 +71,30 @@
 else:
     items_get_request_model = ItemCollectionUri
 
+if "collection_search" in _enabled_extensions:
+    collection_search_extension = CollectionSearchExtension.from_extensions(
+        extensions=extensions
+    )
+    collections_get_request_model = collection_search_extension.GET
+else:
+    collection_search_extension = None
+    collections_get_request_model = EmptyRequest
+
 post_request_model = create_post_request_model(extensions, base_model=PgstacSearch)
 get_request_model = create_get_request_model(extensions)
 
 api = StacApi(
     settings=settings,
-    extensions=extensions,
-    client=CoreCrudClient(post_request_model=post_request_model),  # type: ignore
+    extensions=extensions + [collection_search_extension],
```

Comment:
when there is no `collection_search` extension we will have `[None]`. I think we should avoid this